### PR TITLE
Migrate log4j to reload4j

### PR DESCRIPTION
**Pull request by nlou9**
Description
This is a request to migrate from confluent-log4j to reload4j

Jira Ticket:
DP-7573

Resolution
Step 1: Replace slf4j-log4j12 with slf4j-reload4j.jar
Step 2: Remove Confluent-log4j
Step 3: Add dependency on Logredactor

### CODE REVIEW

File: base/pom.xml [hunk 1]
```diff
@@ -76,6 +76,17 @@
             <groupId>io.confluent</groupId>
             <artifactId>disk-usage-agent</artifactId>
             <version>${io.confluent.common.version}</version>
```
[the remark this answers is not in the document]
Comment (gracechensd) [reply base/pom.xml:78]: I think `common` does have a plan to remove `slf4j-log4j12` [described here](https://confluentinc.atlassian.net/wiki/spaces/~913794610/pages/2772764716/Backport+reload4j+to+replace+Confluent-log4j+in+7.1.x+7.2.x+and+master#Define-Versions-in-Common) (driven by Xin Jin) once all other repos have successfully migrated and removed it

[the remark this answers is not in the document]
Comment (acourtneybrown) [reply base/pom.xml:78]: looks like i was missing context that @nlou9 provided in slack (& pointed to the tickets & wiki pages from security).

**Review by acourtneybrown — APPROVED** [review]

File: utility-belt/pom.xml [hunk 2]
```diff
@@ -219,11 +223,16 @@
             <scope>test</scope>
         </dependency>
 
-         <dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-reload4j</artifactId>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (nlou9) [reply utility-belt/pom.xml:229]: Add it in base now

**Review by gracechensd — APPROVED** [review]
Looks good to me whenever PR build passes

**Pull request merged**